From 649ea2e6bef80ea3fd003dea0f6bbc3ac5458729 Mon Sep 17 00:00:00 2001
Message-Id: <649ea2e6bef80ea3fd003dea0f6bbc3ac5458729.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Tue, 2 Dec 2014 12:19:05 +0530
Subject: [PATCH 06/29] MUST_REBASE [VPG]: drivers/video/adf: Compile time
 definitions for CHV added in Kconfig

To avoid multiple checks of platform, this option in Kconfig is added. To
be used for specific programming for CHV platform.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4443
Change-Id: I244b9237dc718fcda83004c6b695cfc9573f5249
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/Kconfig |   28 +++++++++++++++++++---------
 1 file changed, 19 insertions(+), 9 deletions(-)

diff --git a/drivers/video/adf/Kconfig b/drivers/video/adf/Kconfig
index 8e6b7ae..7e086c0 100644
--- a/drivers/video/adf/Kconfig
+++ b/drivers/video/adf/Kconfig
@@ -25,18 +25,28 @@ config ADF_INTEL
 	help
 	  Enable ADF compatible Intel display driver.
 
-choice
-	prompt "Intel Gen Platform"
+config ADF_INTEL_VLV
+	bool "Valleyview/Cherryview"
 	depends on ADF_INTEL
-	default ADF_INTEL_VLV
+	default y
 	help
-	  The ADF driver supports multiple Intel GEN platforms. Select the targeted one.
+	  Choose this option for common code on VLV/CHT platforms.
 
-	config ADF_INTEL_VLV
-		bool "Valleyview"
-		help
-		  Intel VLV SoC Platform.
-endchoice
+	  Google has come up with a new display framework called ADF. Intel
+	  linux display team has come up with a new driver based on the ADF
+	  framework. This options is for commond code for VLV/CHT in the ADF
+	  compliant driver.
+
+config ADF_INTEL_CHV
+	bool "Only Cherryview specific"
+	depends on ADF_INTEL_VLV
+	help
+	  Choose this option for specific code required for CHT platform.
+
+	  Google has come up with a new display framework called ADF. Intel
+	  linux display team has come up with a new driver based on the ADF
+	  framework. This options is for specific code for CHT in the ADF
+	  compliant driver.
 
 config ADF_INTEL_FBDEV
 	depends on ADF_INTEL && ADF_FBDEV
-- 
1.7.9.5

